pip/qsharp/_qsharp.py

```diff
@@ -349,6 +349,27 @@ def __str__(self) -> str:
     def _repr_html_(self) -> str:
         return self.__data._repr_html_()
 
```

Comment:
It would be very helpful to have another helper here to convert a dense array of amplitudes into a StateDump. Since Q# code itself cannot take a dictionary as a parameter, parameters of state preparation operations, for example, are arrays of amplitudes, and a tool to convert between an array and a sparse dict would help to use `check_eq` quite a lot

Reply:
I think I've got this working as expected, such that a list of real amplitudes can be passed in the same way it is passed into state preparation operations. Here's an example (including using a permissive tolerance):
<img width="604" alt="image" src="https://github.com/microsoft/qsharp/assets/10567287/7cab7724-93d7-4afd-b944-e48a3bcc3f36">

Reply:
Why is the list of amplitudes limited to only real numbers? In the specific app where I used this so far it is, but generally (for example, in homework assignments) it can absolutely have complex numbers as well

Reply:
Ah, good point. I was trying to mimic the inputs in Q# which takes just `Double` for state prep, but I forgot I can count on Python to coerce for me. I changed the type to `complex` in here and confirmed it accepts arrays of `float` just fine.

Reply:
There is another operation in Q# that takes array of Complex, so we need to mimic both I'm afraid :-) But yes, Python is very nice like that

Reply:
Added new test cases to cover arrays of complex.